GetVMByName doesnt return err if VM is not found

this code produce https://github.com/openshift/cluster-api-provider-ovirt/blob/master/pkg/actuators/machine/machine.go#L298-L305 produces this error if it tries to remove a non-existing VM.

```I0418 15:43:38.031508  223901 nodeController.go:36] controllers/NodeController "msg"="NodeController, Reconciling:"  "Node"={"Namespace":"","Name":"ovirt10-d4fr5-worker-848h9"}
I0418 15:43:38.031570  223901 providerIDController.go:32] controllers/ProviderIDController "msg"="ProviderIDController, Reconciling"  "Node"={"Namespace":"","Name":"ovirt10-d4fr5-worker-848h9"}
E0418 15:43:38.145241  223901 controller.go:267] ovirt10-d4fr5-worker-848h9: failed to remove finalizer from machine: Operation cannot be fulfilled on machines.machine.openshift.io "ovirt10-d4fr5-worker-848h9": the object has been modified; please apply your changes to the latest version and try again
E0418 15:43:38.145373  223901 controller.go:304] controller-runtime/manager/controller/machine_controller "msg"="Reconciler error" "error"="Operation cannot be fulfilled on machines.machine.openshift.io \"ovirt10-d4fr5-worker-848h9\": the object has been modified; please apply your changes to the latest version and try again" "name"="ovirt10-d4fr5-worker-848h9" "namespace"="openshift-machine-api" 
I0418 15:43:38.145524  223901 controller.go:175] ovirt10-d4fr5-worker-848h9: reconciling Machine
I0418 15:43:38.257261  223901 controller.go:219] ovirt10-d4fr5-worker-848h9: reconciling machine triggers delete
I0418 15:43:38.350722  223901 controller.go:390] Could not find node from noderef, it may have already been deleted: ovirt10-d4fr5-worker-848h9
I0418 15:43:38.350778  223901 klog.go:22] Testing oVirt engine connection...
I0418 15:43:38.714255  223901 klog.go:22] Completed testing oVirt engine connection.
I0418 15:43:38.714356  223901 klog.go:22] Getting vm name ovirt10-d4fr5-worker-848h9...
I0418 15:43:38.815057  223901 klog.go:22] Completed getting vm name ovirt10-d4fr5-worker-848h9.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x80 pc=0x1b416cf]

goroutine 444 [running]:
github.com/openshift/cluster-api-provider-ovirt/pkg/actuators/machine.(*machineScope).delete(0xc000698640)
    /home/eslutsky/GolandProjects/cluster-api-provider-ovirt/pkg/actuators/machine/machine.go:316 +0x16f
github.com/openshift/cluster-api-provider-ovirt/pkg/actuators/machine.(*OvirtActuator).Delete(0xc00014be80, {0x23cab58, 0xc000bb5ec0}, 0xc000d62400)
    /home/eslutsky/GolandProjects/cluster-api-provider-ovirt/pkg/actuators/machine/actuator.go:145 +0x8f
github.com/openshift/machine-api-operator/pkg/controller/machine.(*ReconcileMachine).Reconcile(0xc0003cefa0, {0x23cab58, 0xc000bb5ec0}, {{{0xc000694fa8, 0x1fcb920}, {0xc0007a80c0, 0xc0005aa000}}})
    /home/eslutsky/GolandProjects/cluster-api-provider-ovirt/vendor/github.com/openshift/machine-api-operator/pkg/controller/machine/controller.go:232 +0x1bad
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0003bbe00, {0x23caab0, 0xc00070c400}, {0x1ea0fe0, 0xc00063ab60})
    /home/eslutsky/GolandProjects/cluster-api-provider-ovirt/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:298 +0x303
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0003bbe00, {0x23caab0, 0xc00070c400})
    /home/eslutsky/GolandProjects/cluster-api-provider-ovirt/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:253 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
    /home/eslutsky/GolandProjects/cluster-api-provider-ovirt/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:214 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
    /home/eslutsky/GolandProjects/cluster-api-provider-ovirt/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:210 +0x354```

--- a/vm_get_by_name.go
+++ b/vm_get_by_name.go
@@ -18,21 +18,14 @@ func (o *oVirtClient) GetVMByName(name string, retries ...RetryStrategy) (result
 			}
 			for _, sdkObject := range response.MustVms().Slice() {
 				if mName, ok := sdkObject.Name(); ok {
+					// We re-scan for the name here since the search function may result other VMs too.
 					if name == mName {
 						result, err = convertSDKVM(sdkObject, o)
-						if err != nil {
-							return wrap(
-								err,
-								EBug,
-								"failed to convert vm %s",
-								name,
-							)
-						}
-
+						return err
 					}
 				}
 			}
-			return nil
+			return newError(ENotFound, "No VM found with name %s", name)
 		})
 	return result, err
 }
@@ -45,5 +38,5 @@ func (m *mockClient) GetVMByName(name string, _ ...RetryStrategy) (result VM, er
 			return vm, nil
 		}
 	}
-	return nil, newError(ENotFound, "vm with Name %s not found", name)
+	return nil, newError(ENotFound, "No VM found with name %s", name)
 }

